htdocs/langs/en_US/stocks.lang

```diff
@@ -185,6 +185,7 @@ inventoryReadPermission=View inventories
 inventoryWritePermission=Update inventories
 inventoryValidatePermission=Validate inventory
 inventoryDeletePermission=Delete inventory
+stockAvailabilityReadPermission=See stock availability
```

Comment:
What is the use case to have a user able to read the stock but without knowing stock availability of a product ?

Reply:
True, an user with stock read availability should have its availability reading right checked automatically. Is there a way to do that? I guess through a trigger?

But the inverse is not true, I want some external user to be able to tell if there is some stock available without being able to have the exact stock numbers (as its private). 

The real use case behind this, is by using the REST API, I want to prevent product ordering from a customer (identified as an external user) by hiding the "add to cart" if there is no stock available, but I don't want a clever user to make the REST API call on its own using its key and see the stock numbers other than "available". The product should still be displayed though for later reference.

Do you think there might be a better way around this?

Cheers

Reply:
As an example:

![image](https://user-images.githubusercontent.com/6573376/140649022-9790d7d4-5981-4426-805c-8fc67aa2eced.png)

Reply:
As an example:

![image](https://user-images.githubusercontent.com/6573376/140649049-cf5158a6-fcdb-4a8d-a8c2-39e574b7ffbc.png)

Reply:
As an example:

![image](https://user-images.githubusercontent.com/6573376/140649062-2025490e-806e-404f-a2a4-2d31eb882e76.png)